src/app/reporting/ReportScheduler.h

```diff
@@ -82,25 +82,34 @@ class ReportScheduler : public ReadHandler::Observer, public ICDStateObserver
     /**
      * @class ReadHandlerNode
      *
-     * @brief This class is in charge of determining when a ReadHandler is reportable depending on the monotonic timestamp of the
-     * system and the intervals of the ReadHandler. It inherits the TimerContext class to allow it to be used as a context for a
-     * TimerDelegate so the TimerDelegate can call the TimerFired method when the timer expires.
+     * @brief This class is responsible for determining when a ReadHandler is reportable depending on the monotonic timestamp of
+     *  the system and the intervals of the ReadHandler. It inherits the TimerContext class to allow it to be used as a context for
+     *  a TimerDelegate so that the TimerDelegate can call the TimerFired method when the timer expires.
      *
-     *  The Logic to determine if a ReadHandler is reportable at a precise timestamp is as follows:
-     *  1: The ReadHandler is in the CanStartReporting state
-     *  2: The minimal interval since last report has elapsed
-     *  3: The maximal interval since last report has elapsed or the ReadHandler is dirty
-     *  If the three conditions are met, the ReadHandler is reportable.
+     *  Three conditions that can prevent the ReadHandler from being reportable:
+     *  1: The ReadHandler is not in the CanStartReporting state:
+     *      This condition can be removed by setting the CanStartReporting flag on the ReadHandler
```

Comment:
here and the other conditions below:

Should it rather say something along the lines of `The condition is met` or `is resolved `?